server/cluster/cluster.go

```diff
@@ -126,7 +82,10 @@ func (s *Server) Create(ctx context.Context, q *cluster.ClusterCreateRequest) (*
 		return nil, err
 	}
 
-	c.ConnectionState = appv1.ConnectionState{Status: appv1.ConnectionStatusSuccessful}
+	c.ConnectionState = appv1.ConnectionState{
+		Status:     appv1.ConnectionStatusSuccessful,
+		ModifiedAt: &v1.Time{Time: time.Now()},
+	}
```

Comment:
Since we are persisting connection state in secret, we should store cluster version at the creation time as well. Please use `s.kubectl.GetServerVersion(c.RESTConfig())` instead of `kube.TestConfig(q.Cluster.RESTConfig())` to validate cluster connectivity and get k8s version at the same time. And store the version in the cluster.